### ISSUE TITLE
[1.12] Storage Scanner duplicates items in compacting drawer.

### ISSUE THREAD

**Issue by billybricker** [McJty/RFTools#1278]
I'm using the storage scanner with Xnet and from compacting drawers from Storage Drawers to store some of my things and I noticed thatthe ammount if iron I had in my compacting drawer kept rising even though I hadn't processed any. I started doing some testing and found that when pulling a stack of iron ingots from a compacting drawer is produces a full stack but only removes 7-8 from the drawer.

I've tested a bit in my test pack with only a few mods installed to make sure that's what was happening and saw the same behavior with iron ingots and sometimes it's not pulling any blocks from the drawer even though it's adding them to my inventory. From what I can tell it's only happening with the compacting drawers, all other drawers seem to work properly.

Forge: 14.21.12406
McJtyLib: 1.12-2.4.3
RFTools: 1.12-6.10beta4
Storage Drawers:  1.12-5.2.7

**Comment by McJty** [comment]
I will check it out. Did you also report this to Storage Drawers? Might be nice to get their input on this too

**Comment by billybricker** [comment]
I didn't report it to Storage Drawers, but I can if you want me to, because they seem to be functioning as intended. I can pull things of the the drawers by hand and everything works as intended. It's only when pulling things from a compacting drawer using the storage scanner that the dupe occurs, at least as far as I can tell.

**Comment by billybricker** [comment]
Well I'm going to post this to Storage Drawers issue tracker because a similar thing is happening with Refined Storage.

**Comment by jaquadro** [comment]
Yeah it's almost certainly something on my side.  Still kind of reeling from a total rewrite of the storage core and API. :(
